internal/resources/grafana/resource_alerting_rule_group.go

```diff
@@ -363,11 +363,26 @@ func putAlertRuleGroup(ctx context.Context, data *schema.ResourceData, meta inte
 
 	packedRules := data.Get("rule").([]interface{})
 	rules := make([]*models.ProvisionedAlertRule, 0, len(packedRules))
+
 	for i := range packedRules {
 		rule, err := unpackAlertRule(packedRules[i], group, folder, orgID)
 		if err != nil {
 			return diag.FromErr(err)
 		}
+
+		// Check if a rule with the same name already exists.
+		for _, r := range rules {
+			if *r.Title == *rule.Title {
+				return diag.Errorf("rule with name %q is defined more than once", *rule.Title)
+			}
+		}
+
+		for _, r := range respAlertRules.Payload {
+			if r.UID != rule.UID && *r.Title == *rule.Title && *r.FolderUID == *rule.FolderUID {
+				return diag.Errorf("rule with name %q already exists in the alert group", *rule.Title)
+			}
+		}
```

Comment:
I'm not sure this is useful at all. The API returns a pretty clear error message when there are duplicated rules

Reply:
I tried to use the same rule name under two different alert groups.
It returned an error: 
```
Error: [PUT /v1/provisioning/folder/{FolderUID}/rule-groups/{Group}] PutAlertRuleGroup (status 500): {}
```
on Grafana 9.5.19.

Reply:
Hmm, gotcha. Later versions have better error handling it seems